metis-normalization/src/main/java/eu/europeana/normalization/normalizers/CleanIRIViolationsNormalizer.java

```diff
@@ -0,0 +1,95 @@
+package eu.europeana.normalization.normalizers;
+
+import eu.europeana.normalization.util.Namespace;
+import eu.europeana.normalization.util.XpathQuery;
+import java.util.ArrayList;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.jena.iri.IRI;
+import org.apache.jena.iri.IRIFactory;
+import org.apache.jena.iri.Violation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URISyntaxException;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+
+
+/**
+ * This normalizer cleans, or 'escapes', media link values, replacing all IRI Violation characters
+ * with their respective encoding character.
+ *
+ * @author Joana Sousa
+ */
+
+public class CleanIRIViolationsNormalizer implements ValueNormalizeAction {
+
+  private static final Logger LOG = LoggerFactory.getLogger(CleanIRIViolationsNormalizer.class);
+
+  private static final Namespace.Element RDF_RESOURCE = Namespace.RDF.getElement("resource");
+  private static final Namespace.Element RDF_ABOUT = Namespace.RDF.getElement("about");
+  private static final Namespace.Element ORE_AGGREGATION = Namespace.ORE.getElement("Aggregation");
+  private static final Namespace.Element EDM_IS_SHOWN_BY = Namespace.EDM.getElement("isShownBy");
+  private static final Namespace.Element EDM_IS_SHOWN_AT = Namespace.EDM.getElement("isShownAt");
+  private static final Namespace.Element EDM_OBJECT = Namespace.EDM.getElement("object");
+  private static final Namespace.Element EDM_HAS_VIEW = Namespace.EDM.getElement("hasView");
+  private static final Namespace.Element EDM_WEB_RESOURCE = Namespace.EDM.getElement("WebResource");
+
+  private final XpathQuery RESOURCE_IS_SHOWN_BY_QUERY = createResourceQueries(EDM_IS_SHOWN_BY);
```

Comment:
Those field should be static right?

Reply:
Yup, right, this is a mistake, will fix that